scrapy/commands/genspider.py

```diff
@@ -135,6 +112,32 @@ def _list_templates(self):
             if filename.endswith('.tmpl'):
                 print("  %s" % splitext(filename)[0])
 
+    def _spider_exists(self, name):
+        if not self.settings.get('NEWSPIDER_MODULE'):
+            # if run as a standalone command and file with same filename already exists
+            if exists(name + ".py"):
+                print("%s already exists" % (abspath(name + ".py")))
+                return True
+        else:
+            try:
+                spidercls = self.crawler_process.spider_loader.load(name)
+            except KeyError:
+                pass
+            else:
+                # if spider with same name exists
+                print("Spider %r already exists in module:" % name)
+                print("  %s" % spidercls.__module__)
+                return False
```

Comment:
```suggestion
                return True
```

I don’t understand why the tests are not breaking, though. Maybe I am missing something.

If the tests should be breaking with the current code, please first make sure they do, and _then_ change the return value.

Reply:
It looks like the [test](https://github.com/scrapy/scrapy/pull/4623/files#diff-59262c12118a93348565a41ab7f6ce42R168-R169) for this case only asserts the output, and because both "Spider %r already exists in module" and "Created spider %r using template %r " were being displayed it passed the test. On changing it to `return True` only the former will be displayed.
Apologies, can't believe I missed it.

Reply:
If you had not made a mistake with the return value here, we would probably have not detected the issue in the test coverage. Now you can improve the test :slightly_smiling_face:

Reply:
I think we could check the modification time of the file before and after the command is run to ensure the file wasn't rewritten on top of asserting the output to make the test stronger. What do you think? The `os.path` module in python has a method `os.path.getmtime` to get the modification time, so shouldn't be too overbearing.

Reply:
The tests fail as expected now

Reply:
@Gallaecio could you please have a look at the code and tell me what you think when possible